Islandora-CLAW/islandora
# Versioning

**GitHub Issue**: Related to: (https://github.com/Islandora-CLAW/CLAW/issues/740)

# What does this Pull Request do?
* Allows versioning of an object to be initiated from Drupal node updates


# What's new?
* Add a isNewVersion boolean value to the event object that gets sent to Alpaca


# How should this be tested?
* pull in the PR to your islandora modules
* make sure that the event type is included in the emitted event object (could debug from EventGenerator or update an alpaca feature to try this)

# Additional Notes:
One important note is that using the IndexReaction for triggering the create_node_version_in_fedora means that the event is triggered as part of islandora_node_insert and islandora_node_update (i think...). Having it be part of node_update means that the version is created before the node is updated. 
Another way to do this is to use a new reaction (maybe called VersionReaction) that could then be included in any hooks in `islandora.module` although probably islandora_node_update makes the most sense.

# Interested parties
@Islandora-CLAW/committers
